src/lib/npem.c

```diff
@@ -194,28 +196,30 @@ static void npem_wait_command(struct pci_dev *pdev)
 
 char *npem_get_path(const char *cntrl_path)
 {
-	return str_dup(cntrl_path);
+	return strdup(cntrl_path);
 }
 
-enum ibpi_pattern npem_get_state(struct slot_property *slot)
+enum led_ibpi_pattern npem_get_state(struct slot_property *slot)
 {
 	u32 reg;
 	struct pci_dev *pdev = NULL;
 	const struct ibpi2value *ibpi2val;
 	const char *path = slot->slot_spec.cntrl->sysfs_path;
 	struct pci_access *pacc = get_pci_access();
+	struct led_ctx *ctx = slot->bl_device->cntrl->ctx;
```

Comment:
this functionality also operates on empty slots, in which case `bl_device` will not be assigned.
I see that `ctx` can be achieved by controller `slot->slot_spec.cntrl->ctx;`

Reply:
Done, thanks!